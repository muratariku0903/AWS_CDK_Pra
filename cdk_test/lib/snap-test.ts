import * as cdk from 'aws-cdk-lib'
import { Construct } from 'constructs'
import * as s3 from 'aws-cdk-lib/aws-s3'

export class S3Stack extends cdk.Stack {
  constructor(scope: Construct, id: string) {
    super(scope, id)

    new s3.Bucket(this, 'sampleID', {
      bucketName: '2023-09-24-sample-bucket',
      versioned: true,
    })
  }
}
