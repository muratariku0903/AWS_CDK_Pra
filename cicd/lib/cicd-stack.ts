import * as cdk from 'aws-cdk-lib'
import { Construct } from 'constructs'
import { CodePipeline, CodePipelineSource, ShellStep } from 'aws-cdk-lib/pipelines'

export class CicdStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props)

    // aws ci-cd project
    new CodePipeline(this, 'demopipeline', {
      pipelineName: 'murapipe',
      synth: new ShellStep('Synth', {
        input: CodePipelineSource.gitHub('muratariku0903/AWS_CDK_Pra', 'master', {
          authentication: cdk.SecretValue.secretsManager('github-pipeline-token', { jsonField: 'github-token' }),
        }),
        commands: ['cd cicd', 'npm ci', 'npm run build', 'npx cdk synth'],
        primaryOutputDirectory: 'cicd/cdk.out',
      }),
    })
  }
}
