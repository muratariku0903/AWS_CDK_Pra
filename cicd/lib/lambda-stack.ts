import * as cdk from 'aws-cdk-lib'
import { Construct } from 'constructs'
import * as lambda from 'aws-cdk-lib/aws-lambda'
import path = require('path')

export class LambdaStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props)

    new cdk.aws_lambda_nodejs.NodejsFunction(this, 'murataPipeLambda', {
      runtime: lambda.Runtime.NODEJS_18_X,
      handler: 'handler',
      entry: path.join(__dirname, '../handlers/handler.ts'),
    })
  }
}
