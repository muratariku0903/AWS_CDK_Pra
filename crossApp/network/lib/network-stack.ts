import * as cdk from 'aws-cdk-lib'
import { Construct } from 'constructs'
import * as ec2 from 'aws-cdk-lib/aws-ec2'

export class NetworkStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props)

    new ec2.Vpc(this, 'tmpVpc', {
      vpcName: 'tmpVpc1',
      ipAddresses: ec2.IpAddresses.cidr('10.0.0.0/16'),
      natGateways: 0,
    })
  }
}
