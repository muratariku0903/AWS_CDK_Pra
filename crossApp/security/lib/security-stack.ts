import * as cdk from 'aws-cdk-lib'
import { Construct } from 'constructs'
import * as iam from 'aws-cdk-lib/aws-iam'

export class SecurityStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props)

    const role = new iam.Role(this, 'tmpRoleId', {
      roleName: 'ec2Role',
      description: 'role for EC2',
      assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com'),
    })
    role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonS3FullAccess'))

    new cdk.CfnOutput(this, 'exportTmpRoleId', {
      value: role.roleArn,
      description: 'iam role for ec2',
      exportName: 'exportedIamRoleForEC2',
    })
  }
}
