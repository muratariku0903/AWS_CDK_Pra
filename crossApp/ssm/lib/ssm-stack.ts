import * as cdk from 'aws-cdk-lib'
import { Construct } from 'constructs'
import * as ssm from 'aws-cdk-lib/aws-ssm'
// import * as sqs from 'aws-cdk-lib/aws-sqs';

export class SsmStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props)

    new ssm.StringParameter(this, 'tmpSsmParameters', {
      parameterName: '/vpc/vpcId',
      stringValue: 'vpc-0fe46a5698d534422',
    })
  }
}
