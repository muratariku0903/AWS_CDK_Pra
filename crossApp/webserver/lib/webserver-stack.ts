import * as cdk from 'aws-cdk-lib'
import { Construct } from 'constructs'
import * as ec2 from 'aws-cdk-lib/aws-ec2'
import { readFileSync } from 'fs'
import * as ssm from 'aws-cdk-lib/aws-ssm'
import * as iam from 'aws-cdk-lib/aws-iam'

export class WebserverStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, {
      ...props,

      // set default env to exec valueFromLookup method
      env: {
        account: '262115391162',
        region: 'ap-northeast-1',
      },
    })

    // get vpc id form ssm parameter
    const vpcId = ssm.StringParameter.valueFromLookup(this, '/vpc/vpcId')

    const vpc = ec2.Vpc.fromLookup(this, 'newVpc', { vpcId: vpcId })
    const sg = new ec2.SecurityGroup(this, 'tmpSg', {
      vpc: vpc,
      securityGroupName: 'allow http traffic',
      allowAllOutbound: true,
    })
    sg.addIngressRule(ec2.Peer.anyIpv4(), ec2.Port.tcp(80), 'allow http traffic')

    // import role security stack
    const role = iam.Role.fromRoleArn(this, 'IAMRoleImport', cdk.Fn.importValue('exportedIamRoleForEC2'))

    const demoEc2 = new ec2.Instance(this, 'tmpEC2', {
      vpc: vpc,
      vpcSubnets: { subnetType: ec2.SubnetType.PUBLIC },
      securityGroup: sg,
      role: role,
      instanceType: ec2.InstanceType.of(ec2.InstanceClass.T2, ec2.InstanceSize.MICRO),
      machineImage: new ec2.AmazonLinuxImage({ generation: ec2.AmazonLinuxGeneration.AMAZON_LINUX_2 }),
      keyName: 'murata-key-02',
    })

    const userData = readFileSync('./lib/userdata.sh', 'utf-8')
    demoEc2.addUserData(userData)
  }
}
