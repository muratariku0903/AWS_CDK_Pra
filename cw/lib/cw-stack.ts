import * as cdk from 'aws-cdk-lib'
import { Construct } from 'constructs'
import * as s3 from 'aws-cdk-lib/aws-s3'
import * as ec2 from 'aws-cdk-lib/aws-ec2'

export class CwStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props)

    // create an s3 bucket
    const bucket = new s3.Bucket(this, 'CWS3', {
      versioned: true,
      bucketName: 'cw-s3',
      removalPolicy: cdk.RemovalPolicy.DESTROY,
      autoDeleteObjects: true,
    })

    // create a vpc
    const vpc = new ec2.Vpc(this, 'CwVpc', {
      natGateways: 0,
      vpcName: 'CwVpc',
      subnetConfiguration: [
        {
          cidrMask: 24,
          name: 'ingress',
          subnetType: ec2.SubnetType.PUBLIC,
        },
        {
          cidrMask: 24,
          name: 'application',
          subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS,
        },
        {
          cidrMask: 28,
          name: 'rds',
          subnetType: ec2.SubnetType.PRIVATE_ISOLATED,
        },
      ],
    })

    // create an RDS instance
    const rds = new cdk.aws_rds.DatabaseInstance(this, 'CwRds', {
      engine: cdk.aws_rds.DatabaseInstanceEngine.MYSQL,
      vpc: vpc,
      databaseName: 'CwRds',
      vpcSubnets: {
        subnetType: ec2.SubnetType.PRIVATE_ISOLATED,
      },
    })
  }
}
