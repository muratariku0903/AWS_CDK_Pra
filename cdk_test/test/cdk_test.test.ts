import * as cdk from 'aws-cdk-lib'
import { Match, Template } from 'aws-cdk-lib/assertions'
import * as CdkTest from '../lib/cdk_test-stack'
import { CdkTestStack } from '../lib/cdk_test-stack'
import { S3Stack } from '../lib/snap-test'

// assertion test
// test('S3 test', () => {
//   const app = new cdk.App()
//   // WHEN
//   const stack = new CdkTest.CdkTestStack(app, 'MyTestStack')
//   // THEN
//   const template = Template.fromStack(stack)
//   template.hasResourceProperties('AWS::S3::Bucket', {
//     BucketName: '2023-09-24-sample-bucket',
//     VersioningConfiguration: {
//       Status: 'Enabled',
//     },
//   })
// })

// snapshot test
describe('snapshot test', () => {
  test('s3 snapshot test', () => {
    const app = new cdk.App()
    const stack = new S3Stack(app, 'sampleID')

    const template = Template.fromStack(stack)
    expect(template.toJSON()).toMatchSnapshot()
  })
})
