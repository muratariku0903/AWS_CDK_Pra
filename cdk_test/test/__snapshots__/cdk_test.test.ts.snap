// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test s3 snapshot test 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "sampleIDF5708235": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketName": "2023-09-24-sample-bucket",
        "VersioningConfiguration": {
          "Status": "Enabled",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
